src/target/source/source_module.cc

```diff
@@ -252,7 +252,9 @@ class CSourceCrtMetadataModuleNode : public runtime::ModuleNode {
           << ") {";
     code_ << "return " << run_func << "(";
     for (const auto& input : metadata_->inputs) {
-      code_ << "inputs->" << input << ",";
+      std::string sanitised_input = input;
+      std::replace_if(sanitised_input.begin(), sanitised_input.end(), ::ispunct, '_');
```

Comment:
is it sufficient just to replace if something is punctuation? what if it's non-alphanumeric?

Reply:
Good point @areusch . I'll provide a fix for that.